### ISSUE TITLE
Cascading Worldgen Lag (Silly Forge)

### ISSUE THREAD

**Issue by Keaft** [AtomicStryker/atomicstrykers-minecraft-mods#233]
#135 #131 #100 

`[Server thread/WARN]: Ruins Mod loaded a new chunk [23, -14] in dimension 0 (overworld) while populating chunk [23, -13], causing cascading worldgen lag.`
`[Server thread/WARN]: Please report this to the mod's issue tracker. This log can be disabled in the Forge config.`

It's kinda silly that FML would say to report this without giving some sort of indication on how to approach improving worldgen times. Should be something like ` Please report this to the mod's issue tracker [url to best practices]` but I digress. 

https://www.reddit.com/r/feedthebeast/comments/5x0twz/investigating_extreme_worldgen_lag/

Basically all about snapping the structure generation to the 16x16 grid with a +8, +8 offset to create a buffer ensuring it doesn't spill over into another chunk. Thank you for making Battle Towers to where you can specify the distance between structures. Shouldn't that make it possible for me to manually snap them to the 16x16 grid correctly? Does the ruins mod also have that ability? Or am I way off in left field and this isn't really possible under current circumstances?

Your mods are absolutely awesome! The diversity that they bring to the game are invaluable!

**Comment by liach** [comment]
@keaft it is not actually a lag; it is just make the game generate a few more chunks when generating the world. Fixing this issue may ensure the safety of world and chunk loading.

**Comment by Keaft** [comment]
You know what. Battle towers is pittance and pennies when it comes to server resources being devoted to generating the world. At most the battle tower will spawn 3 chunks, which is like nothing compared to some of the other mods. Ruins is mostly the same with a few exceptions. Either way it's nothing like this crazy example from the above link: http://i.imgur.com/SH3yRGa.png

I'm still kinda curious if it's possible to manually offset to ensure the least number of extra chunks loaded.

So if the structure offset was already know to you, and this issue being negligible anyway, please feel free to close this and take my apologies for bringing it up again.

**Comment by AtomicStryker** [comment]
You can count the instances of "Ruins Mod loaded a new chunk" in your logfile?

No, it is not possible to optimize for this. Ruins takes a random coordinate in a spawned chunk for *configured tries* amount of attempts and checks a template for allowing the given blocks and ground unevenness before placing it.

Forge just logs this because people complain to them about slow worldgen when it is actually mods like Ruins that cause these cascades - which can really go out of control like your image shows, but only if several aggressively generating mods work in tandem.
